vector/v.neighbors/main.c

```diff
@@ -54,125 +103,281 @@ int main(int argc, char *argv[])
 
-	    box.E = x + radius;
-	    box.W = x - radius;
+            x = Rast_col_to_easting(col + 0.5, &region);
 
-	    Vect_select_lines_by_box(&In, &box, GV_POINTS, List);
-	    G_debug(3, "  %d lines in box", List->n_values);
+            box.E = x + radius;
+            box.W = x - radius;
 
-	    count = 0;
+            Vect_select_lines_by_box(&In, &box, GV_POINTS, List);
+            G_debug(3, "  %d lines in box", List->n_values);
 
-	    for (i = 0; i < List->n_values; i++) {
-		Vect_read_line(&In, Points, Cats, List->id[i]);
+            count = 0;
 
-		if (field != -1 && Vect_cat_get(Cats, field, NULL) == 0)
-		    continue;
-		
-		if (Vect_points_distance(x, y, 0.0, Points->x[0],
-					 Points->y[0], 0.0, 0) <= radius)
-		    count++;
-	    }
+            for (i = 0; i < List->n_values; i++) {
 
-	    if (count > 0) {
-		value = count;
-		Rast_set_d_value(rp, value, CELL_TYPE);
-	    }
-	    rp = G_incr_void_ptr(rp, Rast_cell_size(CELL_TYPE));
-	    count_sum += count;
-	}
+                Vect_read_line(&In, Points, Cats, List->id[i]);
+
+                if (field != -1 && Vect_cat_get(Cats, field, NULL) == 0)
+                    continue;
+
+                if (field > 0 &&
+                    !Vect_cats_in_constraint(Cats, field, pcat_list))
+                    continue;
+
+
+                if (Vect_points_distance(x, y, 0.0, Points->x[0],
+                                         Points->y[0], 0.0, 0) <= radius) {
+
+                    count++;
+
+                    if (strcmp(method_opt->answer, "count") != 0) {
 
-	Rast_put_row(out_fd, result, CELL_TYPE);
+                        if (Cats->n_cats > 1)
+                            G_warning(_("Several cat values found for point %d. Using only first"),
+                                      List->id[i]);
+                        tmp_cat = Cats->cat[0];
+                        /* find cat in array */
+                        db_CatValArray_get_value(&cvarr, tmp_cat, &catval);
+
+                        if (catval) {
+                            switch (cvarr.ctype) {
+                            case DB_C_TYPE_INT:
+                                pvalcats[npvalcats] = catval->val.i;
+                                npvalcats++;
+                                break;
+
+                            case DB_C_TYPE_DOUBLE:
+                                pvalcats[npvalcats] = catval->val.d;
+                                npvalcats++;
+                                break;
+                            }
+                            if (npvalcats >= npvalcatsalloc) {
+                                npvalcatsalloc += 10;
+                                pvalcats =
+                                    (double *)G_realloc(pvalcats,
+                                                        npvalcatsalloc
+                                                        * sizeof(double));
+                            }
+                        }
+
```

Comment:
Please remove too many empty lines, also above and below.

Reply:
I tried to reduce the number of empty lines. Is there any official policy on that ?